modules/internal/ArrayViewRankChange.chpl

```diff
@@ -357,7 +357,18 @@ module ArrayViewRankChange {
       return downDom.dsiHasSingleLocalSubdomain();
 
     proc dsiLocalSubdomain() {
-      return downDom.dsiLocalSubdomain();
+      const dims = downDom.dsiLocalSubdomain().dims();
+      const empty : domain(rank, idxType, chpl__anyStridable(dims));
+
+      // If the rank-changed dimension's index is not a member of the range
+      // in the same dimension of 'dims', then this locale does not have a
+      // local subdomain.
+      for param d in 1..dims.size {
+        if collapsedDim(d) && !dims(d).member(idx(d)) then
+          return empty;
+      }
```

Comment:
Thinking through this: for a ``{3, ..}`` rank-change slicing dom on a block-distributed array/domain, the down domain would be ``{3..3, ..}`` in which case the processors owning column 3 would see ``{3..3, <whatever>}`` as their local down-subdomain and the others would see ``{1..0, <whatever>}`` as theirs.  So rather than doing a membership check, could we simply see if dims(d) is empty?  (though I guess the two are arguably equivalent, this way just feels slightly more convoluted... I'd guess it's asking "Is 3 a member of this empty range?"